components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/GoogleOAuth2Authenticator.java

```diff
@@ -334,7 +445,63 @@ protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse tok
         } catch (IOException e) {
             log.error("Communication error occurred while accessing user info endpoint", e);
         }
-
         return claims;
     }
+
+    /**
+     * This function validates the CSRF double-sided cookie returned from Google One Tap respond.
+     * The request is considered as non-attacked request if the CSRF cookie and the parameter is equal.
+     *
+     * @param request HttpServletRequest. Authentication request with Google One Tap auth payloads.
+     * @return Integrity of the authentication request sent via Google One Tap.
+     */
+    private boolean validateCSRFCookies(HttpServletRequest request) {
+
+        if (request.getCookies() == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("No valid cookie found for Google One Tap authentication.");
+            }
+            return false;
+        }
+        List<Cookie> crossRefCookies = Arrays.stream(request.getCookies())
+                .filter(cookie -> cookie.getName().equalsIgnoreCase(G_CSRF_TOKEN))
+                .collect(Collectors.toList());
```

Comment:
Can't we use `.findFirst().orElse(null)` if there is only one value?

Reply:
Fixed 3335ebd823fbdae6364b8cc712db7975815b8ee6